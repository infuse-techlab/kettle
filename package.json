{
  "name": "kettle",
  "version": "0.1.0",
  "description": "CLI multi usage application handler",
  "main": "index.js",
  "repository": "https://github.com/infuse-techlab/kettle.git",
  "author": "Julien Andreu <julienandreu@me.com>",
  "license": "MIT",
  "private": false,
  "devDependencies": {
    "@tsconfig/recommended": "^1.0.1",
    "@types/jest": "^26.0.16",
    "@types/node": "^14.14.10",
    "@typescript-eslint/eslint-plugin": "^4.9.0",
    "@typescript-eslint/parser": "^4.9.0",
    "eslint": "^7.15.0",
    "eslint-config-prettier": "^6.15.0",
    "eslint-plugin-prettier": "^3.2.0",
    "husky": "^5.0.4",
    "jest": "^26.6.3",
    "lint-staged": "^10.5.3",
    "nodemon": "^2.0.6",
    "pinst": "^2.1.1",
    "prettier": "^2.2.1",
    "ts-jest": "^26.4.4",
    "ts-node": "^9.1.0",
    "typescript": "^4.1.2"
  },
  "scripts": {
    "start": "ts-node --project tsconfig.json src/index.ts",
    "lint": "eslint '*/**/*.{js,ts,tsx}' --fix",
    "test": "jest",
    "test:related": "jest --bail --findRelatedTests",
    "cover": "jest --coverage",
    "quality": "sonar-scanner"
  },
  "dependencies": {
    "commander": "^6.2.0",
    "reflect-metadata": "^0.1.13",
    "typedi": "^0.8.0"
  }
}
