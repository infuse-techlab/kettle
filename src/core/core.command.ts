import { Command } from 'commander';
import { Service } from 'typedi';

@Service()
export class CoreCommand extends Command {}
