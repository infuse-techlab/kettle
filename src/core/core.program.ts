import { Service } from 'typedi';
import { name, version } from '../../package.json';
import { CoreCommand } from './core.command';

@Service()
export class CoreProgram {
  constructor(private readonly command: CoreCommand) {
    this.init();
  }

  private init = () => {
    this.command.name(name);
    this.command.version(version, '-v, --version', 'output the current version');
    this.command.parse();
  };

  // noinspection JSUnusedGlobalSymbols
  public run = async (): Promise<void> => {
    console.log('Hello world!');
  };
}
