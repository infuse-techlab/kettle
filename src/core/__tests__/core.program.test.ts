import { name, version } from '../../../package.json';
import { CoreCommand } from '../core.command';
import { CoreProgram } from '../core.program';

jest.mock('../core.command');

describe('CoreProgram', () => {
  it('should set be instantiated', () => {
    const mockCoreCommand = new CoreCommand();
    const coreProgram = new CoreProgram(mockCoreCommand);
    expect(coreProgram).toBeInstanceOf(CoreProgram);
  });

  it('should set program name', () => {
    const mockCoreCommand = new CoreCommand();
    new CoreProgram(mockCoreCommand);
    expect(mockCoreCommand.name).toBeCalledTimes(1);
    expect(mockCoreCommand.name).toBeCalledWith(name);
  });

  it('should set command version', () => {
    const mockCoreCommand = new CoreCommand();
    new CoreProgram(mockCoreCommand);
    expect(mockCoreCommand.version).toBeCalledTimes(1);
    expect(mockCoreCommand.version).toBeCalledWith(version, '-v, --version', 'output the current version');
  });

  it('should parse arguments', () => {
    const mockCoreCommand = new CoreCommand();
    new CoreProgram(mockCoreCommand);
    expect(mockCoreCommand.parse).toBeCalledTimes(1);
    expect(mockCoreCommand.parse).toBeCalledWith();
  });

  it('should run the app', () => {
    const mockCoreCommand = new CoreCommand();
    const coreProgram = new CoreProgram(mockCoreCommand);
    coreProgram.run();
  });

  it(`should console.log "Hello world!" the app`, () => {
    const mockCoreCommand = new CoreCommand();
    const coreProgram = new CoreProgram(mockCoreCommand);
    const spyConsoleLog = jest.spyOn(console, 'log');
    coreProgram.run();
    expect(spyConsoleLog).toBeCalledTimes(1);
    expect(spyConsoleLog).toBeCalledWith('Hello world!');
    spyConsoleLog.mockRestore();
  });

  it('should have no regression', () => {
    const coreProgram = new CoreCommand();
    expect(coreProgram).toMatchSnapshot();
  });
});
