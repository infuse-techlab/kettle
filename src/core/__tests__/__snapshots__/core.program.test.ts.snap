// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreProgram should have no regression 1`] = `
CoreCommand {
  "name": [MockFunction],
  "parse": [MockFunction],
  "version": [MockFunction],
}
`;
